synapse/handlers/room_member.py

```diff
@@ -835,7 +843,7 @@ def _lookup_3pid_v2(
                 supported_lookup_algorithms,
             )
             raise SynapseError(
-                400,
+                500,
```

Comment:
I think we decided this should be a 400 too? In any case, don't use SynapseError for 500s.